e2e/tests/dashboard-tests.spec.ts

```diff
@@ -0,0 +1,162 @@
+import { test, expect } from "@playwright/test";
+
+test.describe("Dashboard Tests", () => {
+  test("should display Quick Actions widget on dashboard", async ({ page }) => {
+    // Go to the dashboard page
+    await page.goto("http://localhost:4200");
+
+    // Wait for "Quick Actions" element to load
+    await page.waitForSelector("text=Quick actions");
+
+    // Check if "Quick Actions" element is visible
+    const quickActionsElement = page.locator("text=Quick actions");
+    await expect(quickActionsElement).toBeVisible();
+
+    // Verify that Record Attendance button exists in Quick actions widget
+    const recordAttendanceButton = page.getByRole("cell", {
+      name: "Record Attendance",
+    });
+    await expect(recordAttendanceButton).toBeVisible();
+
+    // Verify children count is displayed
+    const childrenCount = page.locator("app-entity-count-dashboard-widget");
+
+    await expect(childrenCount).toContainText(/107/);
+  });
+
+  test("should navigate to the attendence page", async ({ page }) => {
+    await page.goto("http://localhost:4200/");
+
+    await page.click("text=Attendance");
+
+    // Check if the URL contains "attendance"
+    await expect(page).toHaveURL(/.*attendance/);
+  });
+
+  test("should display tasks due with correct due dates", async ({ page }) => {
+    // Go to the dashboard page
+    await page.goto("http://localhost:4200");
+
+    await page.waitForSelector("text=Tasks due");
+
+    // Check for the "Tasks due" widget
+    const tasksDueElement = page.locator("text=Tasks due");
+
+    await expect(tasksDueElement).toBeVisible();
+
+    // Verify that the task names and due dates are displayed
+    const taskNames = await page
+      .locator("app-widget-content")
+      .filter({ hasText: "get signed agreement Nov 19," })
```

Comment:
Does this test pass, even now (when 19th Nov is a long time in the past)?

Reply:
I have restructured it.